### PR TITLE
Save get

### PR DESCRIPTION
**Pull request by ooookai**

### CODE REVIEW

File: private/jsx/sidebar/Controls.js [hunk 1]
```diff
@@ -1,17 +1,64 @@
...
+                console.log('saveMemory success', { res })
+                Act.setModified({ value: false })
+            })
+            .catch(e => console.error(e))
+            .then(() => {
+                setTimeout(() => {
+                    this.setState({
+                        saving: false,
+                    })
+                }, 1300)
```
Comment (ooookai) [inline private/jsx/sidebar/Controls.js:59]: Since the save function is too faster to trust it is already happened, I set 1.3 secs for loading status.

**Comment by ooookai** [comment]
I haven't test this function when I log in as an User.

File: views/app.jade [hunk 2]
```diff
@@ -5,6 +5,7 @@ append content
 
     script.
         var datavoxelId = #{datavoxelId};
+        var userSignedIn = #{userSignedIn};
```
Comment (ooookai) [inline views/app.jade:8]: access user sign-in information

File: private/jsx/sidebar/Controls.js (hunk 1 above)
Comment (cesandoval) [reply private/jsx/sidebar/Controls.js:59]: the function is not autosaving currently

File: views/app.jade (hunk 2 above)
Comment (cesandoval) [reply views/app.jade:8]: even if you are not logged in, you can still save, which shouldn't be the case!

**Review by cesandoval — CHANGES_REQUESTED** [review]
Hey! the function looks great! it still does not autosave, and show the save icon even if not logged in

File: private/jsx/sidebar/Controls.js (hunk 1 above)
Comment (ooookai) [reply private/jsx/sidebar/Controls.js:59]: you want to apply autosaving??

File: views/app.jade (hunk 2 above)
Comment (ooookai) [reply views/app.jade:8]: what do you mean?

**Comment by ooookai** [comment]
@cesandoval I need you to test the API function when you log-in, and then i will disable the icon when user not log-in.

**Comment by cesandoval** [comment]
The API works when I am logged in. It still does not autosave when the map is loaded though.

**Comment by ooookai** [comment]
OK, I will disable the save button when user isn't logged in. 
I thought you know why I don't want the autosaving function?

**Comment by cesandoval** [comment]
Ok sounds good! Yeah, I thought you had implemented the autosave function, so I got confused. It's fine not to have the autosave for now, I think instead, we can generate the layer colors in the backend and use those by default. But all of that is future work. 

For the save function, I realized that we are actually sending a `json` response once it is saved in the backend. So we can just wait until that response is sent back, instead of the 1300ms delay?

https://github.com/cesandoval/PaintingWithData_Riyadh/pull/248/commits/77d51d16b2f934a6f378fb7693043a3a82b9559c#diff-762420410ec047d5abd72f91dd312730L22

Let me know when I can merge! or just send a PR.

**Comment by ooookai** [comment]
1300ms is the additional delay after the response, in order to let user trust the function is working instead of failing

**Comment by cesandoval** [comment]
ok, cool, sounds good! Once you disable it, when a user is not logged in, we can do a PR!

**Pull request merged**